bytecomp/simplif.ml

```diff
@@ -794,7 +795,9 @@ let simplify_local_functions lam =
     | Lvar id ->
         Hashtbl.remove slots id
     | Lfunction lf as lam ->
-        check_static lf;
+        if Lambda.function_is_curried lf then begin
```

Comment:
I think this chunk is wrong: it removes the warning telling the user that the optimization doesn't apply even though it was requested (and precisely, if the function is tupled, the optimization does not trigger and this should be reported as such).

Reply:
Fixed